firebase_admin/_auth_client.py

```diff
@@ -36,17 +42,45 @@ def __init__(self, app, tenant_id=None):
             2. set the project ID explicitly via Firebase App options, or
             3. set the project ID via the GOOGLE_CLOUD_PROJECT environment variable.""")
 
-        credential = app.credential.get_credential()
+        credential = None
         version_header = 'Python/Admin/{0}'.format(firebase_admin.__version__)
+        # Non-default endpoint URLs for emulator support are set in this dict later.
+        id_toolkit_endpoints = {}
+
+        # If an emulator is present, check that the given value matches the expected format and set
+        # endpoint URLs to use the emulator. Also set a fake authorization token.
+        emulator_host = os.environ.get(_EMULATOR_HOST_ENV_VAR)
+        if emulator_host:
+            if '//' in emulator_host:
+                raise ValueError(
+                    'Invalid {0}: "{1}". It must follow format "host:port".'.format(
+                        _EMULATOR_HOST_ENV_VAR, emulator_host))
+            base_url = 'http://{0}/identitytoolkit.googleapis.com'.format(emulator_host)
+            id_toolkit_endpoints['v1'] = base_url + '/v1'
+            id_toolkit_endpoints['v2beta1'] = base_url + '/v2beta1'
+
+        try:
+            # Use credentials if provided
+            credential = app.credential.get_credential()
+        except DefaultCredentialsError as credential_error:
+            # The emulator can make do with fake credentials
+            if emulator_host:
+                credential = credentials.Credentials(token='owner')
+            else:
+                # Not emulated, no credential, raise
+                raise credential_error
+
         http_client = _http_client.JsonHttpClient(
             credential=credential, headers={'X-Client-Version': version_header})
 
         self._tenant_id = tenant_id
-        self._token_generator = _token_gen.TokenGenerator(app, http_client)
+        self._token_generator = _token_gen.TokenGenerator(
```

Comment:
I'd recommend using keyword args for the last argument for clarity. But not a dealbreaker.